### PR TITLE
Update Helix API generation to use swagger generator

### PR DESCRIPTION
**Pull request by alexperovich**
This change updates the Helix API generation to use a more extensible generator and adds handling for error messages that the server sends.

With this change HTTP 400 responses should be much easier to diagnose.

### CODE REVIEW

**Comment by jonfortescue** [comment]
nit: is it possible to make them still go to `Generated` rather than `generated-code`? Doesn't really matter but

File: Arcade.sln [hunk 1]
```diff
@@ -1,14 +1,12 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio Version 16
-VisualStudioVersion = 16.0.28411.114
+# Visual Studio 15
+VisualStudioVersion = 15.0.28307.271
```
Comment (jonfortescue) [inline Arcade.sln]: is this change intentional?

File: src/HelixPoolProvider/HelixPoolProvider/Controllers/PoolProviderController.cs [hunk 2]
```diff
@@ -116,7 +116,7 @@ private async Task<HelixJobCreator> GetHelixJobCreator(AgentAcquireItem agentReq
                 _logger.LogInformation($"Queue {queueId} does not exist");
                 return null;
             }
-            else if (!queueInfo.IsAvailable.HasValue || !queueInfo.IsAvailable.Value)
```
Comment (jonfortescue) [inline src/HelixPoolProvider/HelixPoolProvider/Controllers/PoolProviderController.cs]: why is this check no longer necessary?

File: src/HelixPoolProvider/HelixPoolProvider/Controllers/PoolProviderController.cs [hunk 3]
```diff
@@ -378,7 +373,7 @@ public async Task<IActionResult> GetAgentDefinition(string agentDefinitionId)
 
                 using (IHelixApi api = GetHelixApi(false))
                 {
-                    var queueInfo = await api.Information.QueueInfoMethodAsync(agentDefinitionId);
+                    var queueInfo = await api.Information.QueueInfoAsync(agentDefinitionId);
```
Comment (jonfortescue) [inline src/HelixPoolProvider/HelixPoolProvider/Controllers/PoolProviderController.cs]: I'm assuming this method name changed as a result of the API switch?

File: src/HelixPoolProvider/HelixPoolProvider/Controllers/PoolProviderController.cs (hunk 2 above)
Comment (jonfortescue) [reply src/HelixPoolProvider/HelixPoolProvider/Controllers/PoolProviderController.cs]: wait I'm guessing it's bc of the API change ignore

File: Arcade.sln (hunk 1 above)
Comment (alexperovich) [reply Arcade.sln]: I changed the solution in VS. I just let it do what it wants to.

File: src/HelixPoolProvider/HelixPoolProvider/Controllers/PoolProviderController.cs (hunk 2 above)
Comment (alexperovich) [reply src/HelixPoolProvider/HelixPoolProvider/Controllers/PoolProviderController.cs]: Ya, these properties aren't nullable anymore.

File: src/HelixPoolProvider/HelixPoolProvider/Controllers/PoolProviderController.cs (hunk 3 above)
Comment (alexperovich) [reply src/HelixPoolProvider/HelixPoolProvider/Controllers/PoolProviderController.cs]: Yep

**Comment by alexperovich** [comment]
The new error messages will be like this for any 400 requests: `System.ArgumentException: Required json property missing: Creator`

File: Arcade.sln (hunk 1 above)
Comment (jonfortescue) [reply Arcade.sln]: lol sounds reasonable

**Comment by jonfortescue** [comment]
nice
